Add symmetric_fill to Slider and use it for ColorPicker intensity slider.

WhalesState opened the pull request:
https://github.com/user-attachments/assets/a28e344d-c5b4-4371-ac8f-99996301624c

Also make the intensity slider respect the center_slider_grabbers theme constant in `ColorPicker`.

<img width="1234" height="985" alt="image" src="https://github.com/user-attachments/assets/9b3e61c7-91b9-4169-8ae8-03346f4c28f9" />

Note: I'd prefer an easier name.

Closes: https://github.com/godotengine/godot-proposals/issues/13153

WhalesState commented on scene/gui/slider.cpp:
```diff
@@ -269,98 +269,121 @@ void Slider::_notification(int p_what) {
 					}
 				}
-				grabber->draw(ci, Point2i(size.width / 2 - grabber->get_width() / 2 + theme_cache.grabber_offset, size.height - ratio * areasize - grabber->get_height() + grabber_shift));
+				if (grabber.is_valid()) {
```
I wonder if this will be useful when users uses empty textures and draws their own custom grabbers, like this

<img width="870" height="518" alt="image" src="https://github.com/user-attachments/assets/4a0e9cc3-4dd5-4fba-8b8a-b28cd9747b36" />

KoBeWi commented on scene/gui/slider.cpp:
```diff
@@ -444,6 +467,37 @@ bool Slider::is_scrollable() const {
+			warnings.push_back(RTR("If \"Symmetric Fill\" is enabled, \"Max Value\" must be greater than 0."));
+		}
+		if (Math::is_equal_approx(get_min(), get_max())) {
+			warnings.push_back(RTR("If \"Symmetric Fill\" is enabled, \"Min Value\" and \"Max Value\" must be different."));
```
Yep.

KoBeWi commented on scene/gui/slider.cpp:
```diff
@@ -269,98 +269,121 @@ void Slider::_notification(int p_what) {
 				style->draw(ci, Rect2i(Point2i(size.width / 2 - widget_width / 2, 0), Size2i(widget_width, size.height)));
-				grabber_area->draw(ci, Rect2i(Point2i((size.width - widget_width) / 2, Math::round(size.height - areasize * ratio - grabber->get_height() / 2 + grabber_shift)), Size2i(widget_width, Math::round(areasize * ratio + grabber->get_height() / 2 - grabber_shift))));
+
+				if (symmetric_fill && get_min() < 0.0 && get_max() > 0.0 && !Math::is_equal_approx(get_min(), get_max())) {
```
Just like with the warning, I don't think the last check is relevant (unless you consider min/max being very close, but not equal to zero, if that can even happen).

WhalesState replied:
We shouldn't call this method when `get_min()` and `get_max()` are approximately equal zero and the previous two checks will only allow this case which can cause division of zero crash.

WhalesState replied:
Maybe we can use `!Math::is_zero_approx(get_max() + get_min())`.

KoBeWi replied:
Eh, the new condition is basically the same.

WhalesState replied:
Without this condition `!Math::is_zero_approx(get_min() - get_max())` the next line may crash 
`const double zero_ratio = get_min() / (get_min() - get_max());`, so we switch to normal fill instead.

WhalesState replied:
Removed it by using CMP_EPSILON instead of 0, and rebased.
`if (symmetric_fill && get_min() < -CMP_EPSILON && get_max() > CMP_EPSILON)`

WhalesState committed:
Add symmetric fill for Slider and use it For ColorPicker intensity slider.

Also make the intensity slider respect the center_slider_grabbers theme constant in ColorPicker.
